### PR TITLE
feat: oauth api login

### PR DESCRIPTION
**Pull request by Mokto**
#### Code-related tasks
- [ ] Tests added for this feature/bug.
- [x] Impact on the seeders.
- [x] Impact on the API.

#### Other tasks
- [x] [CHANGELOG](https://github.com/monicahq/monica/blob/master/CHANGELOG) entry added, if necessary, under `UNRELEASED`.
- [x] [CONTRIBUTORS](https://github.com/monicahq/monica/blob/master/CONTRIBUTORS) entry added, if necessary.

### CODE REVIEW

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:dfW9piseoLLacuF]

**Comment by djaiss** [comment]
@Mokto I think what you've done is good.
But have you recreated this without knowing it https://laravel.com/docs/5.6/passport#requesting-password-grant-tokens ?

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:pEYujHrKjPWQrGF]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:ISgXmphAI28ISx3]

File: app/Http/Controllers/Auth/OAuthController.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    public function __construct(Application $app)
+    {
+        $this->app = $app;
+    }
+
+    /**
+     * Log in a user and returns an accessToken.
+     */
+    public function login(Request $request)
```
Comment (asbiin) [inline app/Http/Controllers/Auth/OAuthController.php]: Code Smell Code Smell: Reduce the number of returns of this function 5, down to the maximum allowed 3. ([php:S1142](https://sonarcloud.io/organizations/monicahq/rules#rule_key=php%3AS1142))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=monica&pullRequest=1174&issues=AWLbqaNJ-pl6AHs2SAyF&open=AWLbqaNJ-pl6AHs2SAyF)

[ISSUE_KEY]: AWLbqaNJ-pl6AHs2SAyF

File: app/Providers/RouteServiceProvider.php [hunk 2]
```diff
@@ -192,6 +193,22 @@ protected function mapWebRoutes(Router $router)
...
+     * @return void
+     */
+    protected function mapOAuthRoutes(Router $router)
+    {
+        $router->group([
+            'prefix' => 'oauth',
+            'namespace' => $this->namespace,
+        ], function ($router) {
+            require base_path('routes/oauth.php');
+        });
```
Comment (asbiin) [inline app/Providers/RouteServiceProvider.php]: Code Smell Code Smell: Remove the unused function parameter "$router". ([php:S1172](https://sonarcloud.io/organizations/monicahq/rules#rule_key=php%3AS1172))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=monica&pullRequest=1174&issues=AWLRaBe7-pl6AHs2IiNi&open=AWLRaBe7-pl6AHs2IiNi)

[ISSUE_KEY]: AWLRaBe7-pl6AHs2IiNi

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:Axz4dhby8l7pqhK]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:CXMoDvwi5bhdX72]

File: app/Http/Controllers/Auth/OAuthController.php (hunk 1 above)
Comment (asbiin) [inline app/Http/Controllers/Auth/OAuthController.php]: Code Smell Code Smell: Reduce the number of returns of this function 5, down to the maximum allowed 3. ([php:S1142](https://sonarcloud.io/organizations/monicahq/rules#rule_key=php%3AS1142))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=monica&pullRequest=1174&issues=AWLbqaNJ-pl6AHs2SAyF&open=AWLbqaNJ-pl6AHs2SAyF)

[ISSUE_KEY]: AWLbqaNJ-pl6AHs2SAyF

File: app/Providers/RouteServiceProvider.php (hunk 2 above)
Comment (asbiin) [inline app/Providers/RouteServiceProvider.php]: Code Smell Code Smell: Remove the unused function parameter "$router". ([php:S1172](https://sonarcloud.io/organizations/monicahq/rules#rule_key=php%3AS1172))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=monica&pullRequest=1174&issues=AWLRaBe7-pl6AHs2IiNi&open=AWLRaBe7-pl6AHs2IiNi)

[ISSUE_KEY]: AWLRaBe7-pl6AHs2IiNi

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:KeunYoNsbS8FOjQ]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:cpL0w9QHNF2rACm]

File: app/Http/Controllers/Auth/OAuthController.php (hunk 1 above)
Comment (asbiin) [inline app/Http/Controllers/Auth/OAuthController.php]: Code Smell Code Smell: Reduce the number of returns of this function 5, down to the maximum allowed 3. ([php:S1142](https://sonarcloud.io/organizations/monicahq/rules#rule_key=php%3AS1142))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=monica&pullRequest=1174&issues=AWLbqaNJ-pl6AHs2SAyF&open=AWLbqaNJ-pl6AHs2SAyF)

[ISSUE_KEY]: AWLbqaNJ-pl6AHs2SAyF

File: app/Providers/RouteServiceProvider.php [hunk 3]
```diff
@@ -192,6 +193,22 @@ protected function mapWebRoutes(Router $router)
...
+     *
+     * @param  \Illuminate\Routing\Router  $router
+     * @return void
+     */
+    protected function mapOAuthRoutes(Router $router)
+    {
+        $router->group([
+            'prefix' => 'oauth',
+            'namespace' => $this->namespace,
+        ], function ($router) {
```
Comment (asbiin) [inline app/Providers/RouteServiceProvider.php]: Code Smell Code Smell: Remove the unused function parameter "$router". ([php:S1172](https://sonarcloud.io/organizations/monicahq/rules#rule_key=php%3AS1172))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=monica&pullRequest=1174&issues=AWLRaBe7-pl6AHs2IiNi&open=AWLRaBe7-pl6AHs2IiNi)

[ISSUE_KEY]: AWLRaBe7-pl6AHs2IiNi

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:vCieXTeBhjYwdqE]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:5wpHHVGqf5xruyW]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:IekBIQstH4eBStX]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:JIBA8zOxDsMruam]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:bMjX3pIz87kAcex]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:YA80WZUqjVk2fUN]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:I2rLSrRB7muvtuc]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:2J9r9vkr7CYuvqW]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:1ZfxWU1Kz4iUVMJ]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:6ADlcm5ozyUdUl2]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:AwNyo75pZOd4nZg]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:SadfUdEDkWZV4jV]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:XYSq20wBntsiaMZ]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:CJ3PBKJOWh0XB9D]

**Comment by asbiin** [comment]
[test]
capture
tests/Browser/screenshots/failure-Tests_Browser_Settings_MultiFAControllerTest_testBarCodeContent-0.png
[id:bHzuOmnVZZWhPKg]

**Pull request merged**